### Resume parser with LLM [llama 3.2 - 3b] via Ollama

In [1]:
# pdf loader for loading / reading the resume

from langchain_community.document_loaders import PyMuPDFLoader

In [2]:
pdf_loader = PyMuPDFLoader('resume_folder\Mahikshith_final_pow-1.pdf')
ans = pdf_loader.load()

In [3]:
print(ans[0].page_content)

Mahikshith
+91-6303423352
mahikshith97@gmail.com
SKILLS
•
Python, SQL, NumPy, Pandas, Matplotlib, Seaborn, Scikit-learn, TensorFlow, Keras, Data Analysis, Stats and Probability, Tableau, 
Hypothesis testing, Machine Learning , NLP, Time series forecasting, Flask, Docker, AWS, EC2 , ECS , SageMaker, LLM , RAG, 
MLflow, Langchain, Web scraping, ETL, CI / CD pipeline
EXPERIENCE
Amazon
Reduced Order Defect Rate (ODR) by 0.27% and brand infringement claims by 9.46% monthly, enhancing operational efficiency
through Amazon QuickSight and SQL analytics.
Developed and implemented ETL pipelines from scratch using Python, increasing data accuracy and timeliness by 7% through
automated ingestion and transformation processes.
Streamlined operational efficiency with CI/CD pipelines, resulting in faster deployment cycles and improved data handling.
Accelerated model development for large datasets with Amazon S3 and AWS SageMaker, achieving a 3% reduction in model
response time in production.
PROJECTS

In [4]:
# Question & Context for LLM : 

context = ans[0].page_content

question = """You are tasked with parsing a resume. Your objective is to extract relevant information in a valid structured 'JSON' format. 
                Do not write explanations or any other preambles."""

In [5]:
# importing the LLm + parser and json parser 
from llm_script.ollama_script import LLM_chain , LLM_chain_json

llm_response = LLM_chain(context=context,question=question)


In [6]:
print(llm_response)

```
{
  "contact": {
    "Name": "Mahikshith",
    "Email": "mahikshith97@gmail.com",
    "Phone Number": "+91-6303423352",
    "Website/Portfolio/LinkedIn": "https://linkedin.com/in/mahikshith",
    "Github Profile": "https://github.com/mahikshith"
  },
  "education": {
    "Institution Name": "Scaler woolf neovarsity",
    "Degree": "MS in Computer Science",
    "Field of Study": "Artificial Intelligence and Machine learning",
    "Graduation Date": "2024"
  },
  "experience": [
    {
      "Job Title": "",
      "Company Name": "Amazon",
      "Location": "",
      "Dates of Employment": "2022-2024",
      "Responsibilities/Projects": "Reduced Order Defect Rate (ODR) by 0.27% and brand infringement claims by 9.46% monthly, enhancing operational efficiency through Amazon QuickSight and SQL analytics."
    }
  ],
  "projects": [
    {
      "Project Title": "AWS-Deployed Predictive Loan Risk classification Model",
      "Description/Technologies Used": "",
      "Outcomes/Results": ""

In [7]:
# let's try json parser 

LLM_chain_json_responze =  LLM_chain_json(llm_response)

In [8]:
print(LLM_chain_json_responze)

{'contact': {'Name': 'Mahikshith', 'Email': 'mahikshith97@gmail.com', 'Phone Number': '+91-6303423352', 'Website/Portfolio/LinkedIn': 'https://linkedin.com/in/mahikshith', 'Github Profile': 'https://github.com/mahikshith'}, 'education': {'Institution Name': 'Scalable Web Learning Factory', 'Degree': 'Master of Science in Computer Science', 'Field of Study': 'Artificial Intelligence and Machine learning', 'Graduation Date': '2024'}, 'experience': [{'Job Title': '', 'Company Name': 'Amazon', 'Location': '', 'Dates of Employment': '2022-2024', 'Responsibilities/Projects': 'Reduced Order Defect Rate (ODR) by 0.27% and brand infringement claims by 9.46% monthly, enhancing operational efficiency through Amazon QuickSight and SQL analytics.'}], 'projects': [{'Project Title': 'AWS-Deployed Predictive Loan Risk classification Model', 'Description/Technologies Used': '', 'Outcomes/Results': ''}, {'Project Title': 'Fine-tuned Transformer-based Text Summarization with NLP & ETL Pipeline implementa

{'contact': {'Name': 'Mahikshith', 'Email': 'mahikshith97@gmail.com', 'Phone Number': '+91-6303423352', 'Website/Portfolio/LinkedIn': 'https://linkedin.com/in/mahikshith', 'Github Profile': 'https://github.com/mahikshith'}, 'education': {'Institution Name': 'Scalable Web Learning Factory', 'Degree': 'Master of Science in Computer Science', 'Field of Study': 'Artificial Intelligence and Machine learning', 'Graduation Date': '2024'}, 'experience': [{'Job Title': '', 'Company Name': 'Amazon', 'Location': '', 'Dates of Employment': '2022-2024', 'Responsibilities/Projects': 'Reduced Order Defect Rate (ODR) by 0.27% and brand infringement claims by 9.46% monthly, enhancing operational efficiency through Amazon QuickSight and SQL analytics.'}], 'projects': [{'Project Title': 'AWS-Deployed Predictive Loan Risk classification Model', 'Description/Technologies Used': '', 'Outcomes/Results': ''}, {'Project Title': 'Fine-tuned Transformer-based Text Summarization with NLP & ETL Pipeline implementation', 'Description/Technologies Used': '', 'Outcomes/Results': ''}], 'skills': [{'Programming Languages': ['Python'], 'Technologies/Tools/frameworks': ['SQL', 'NumPy', 'Pandas', 'Matplotlib', 'Seaborn', 'Scikit-learn', 'TensorFlow', 'Keras', 'Data Analysis', 'Stats and Probability', 'Tableau', 'Hypothesis testing', 'Machine Learning', 'NLP', 'Time series forecasting', 'Flask', 'Docker', 'AWS', 'EC2', 'ECS', 'SageMaker', 'LLM', 'RAG', 'MLflow', 'Langchain', 'Web scraping', 'ETL', 'CI/CD pipeline']}], 'additionalInformation': {'Certifications': '', 'Awards or Honors': '', 'Professional Affiliations': '', 'Languages': ''}}


In [10]:
# storing the json parsed response 

import json 

parsed_file = "mahikshith_resume_parse.json"

json_file = 'output_parsed_resume/{}'.format(parsed_file)

json.dump(LLM_chain_json_responze, open(json_file, 'w'), indent=4)

In [ ]:
# next up --  creating knowledge base